commit bcd92c9fbcc679ee95003083056f0441a1f474fa
Author: Christoph Hellwig <hch@lst.de>
Date:   Thu May 31 20:12:26 2007 +0200

    [SCSI] introduce shost_priv helper
    
    Currently accessing the scsi host private data is rather messy because
    it comes as an unsigned long that always needs a cast first.  This patch
    introduces a helper that does the cast called shost_priv.  It's similar
    in spirit to netdev_priv for network drivers.
    
    This is the first patch introducing the macro, and the second patch
    in the series will convert esp and it's subdrivers as an example.
    Further conversion will wait until the helper is in the tree to make
    patch juggling easier.
    
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: James Bottomley <James.Bottomley@SteelEye.com>

---
 include/scsi/scsi_host.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/scsi/scsi_host.h	2007-07-08 19:32:17.000000000 -0400
+++ b/include/scsi/scsi_host.h	2007-08-27 14:03:20.000000000 -0400
@@ -677,6 +677,10 @@ struct Scsi_Host {
 #define shost_printk(prefix, shost, fmt, a...)	\
 	dev_printk(prefix, &(shost)->shost_gendev, fmt, ##a)
 
+static inline void *shost_priv(struct Scsi_Host *shost)
+{
+	return (void *)shost->hostdata;
+}
 
 int scsi_is_host_device(const struct device *);
 
